openedx/core/djangoapps/schedules/tests/test_signals.py

```diff
@@ -70,19 +72,64 @@ def test_schedule_config_creation_enabled_instructor_paced(self, mock_get_curren
         site = SiteFactory.create()
         mock_get_current_site.return_value = site
         ScheduleConfigFactory.create(site=site, enabled=True, create_schedules=True)
-        course = create_course_run(self_paced=False)
+        course = _create_course_run(self_paced=False)
         enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT)
         with self.assertRaises(Schedule.DoesNotExist):
             enrollment.schedule
 
 
-def create_course_run(self_paced=True):
+@skip_unless_lms
+class UpdateScheduleTests(SharedModuleStoreTestCase):
+    ENABLED_SIGNALS = ['course_published']
+    VERIFICATION_DEADLINE_DAYS = 14
+
+    def setUp(self):
+        super(UpdateScheduleTests, self).setUp()
+        self.site = SiteFactory.create()
+        ScheduleConfigFactory.create(site=self.site)
+        DynamicUpgradeDeadlineConfiguration.objects.create(enabled=True, deadline_days=self.VERIFICATION_DEADLINE_DAYS)
+
+    def assert_schedule_dates(self, schedule, expected_start):
+        self.assertEquals(_strip_secs(schedule.start), _strip_secs(expected_start))
+        self.assertEquals(
+            _strip_secs(schedule.upgrade_deadline),
+            _strip_secs(expected_start) + datetime.timedelta(days=self.VERIFICATION_DEADLINE_DAYS),
+        )
+
+    @patch('openedx.core.djangoapps.schedules.signals.get_current_site')
```

Comment:
nit: I tend to prefer doing this kind of thing in setUp if it's going to be done on every test case.

Reply:
Addressed in https://github.com/edx/edx-platform/pull/15998